node/blockAPI/baseBlock.go

```diff
@@ -151,3 +152,21 @@ func (bap *baseAPIBockProcessor) getBlockStatus(storerUnit dataRetriever.UnitTyp
 
 	return BlockStatusOnChain, nil
 }
+
+func (bap *baseAPIBockProcessor) computeStatusAndPutInBlock(blockAPI *api.Block) (*api.Block, error) {
```

Comment:
storerUnit should come as a parameter. In this way you will avoid this if.

Reply:
done.